### ISSUE TITLE
Refine telecoms layer

### ISSUE THREAD

**Issue by NKAmapper** [openinframap/openinframap#155]
Here is a proposal to make the telecoms layer more visible and useful. The proposal is based on OSM telecom infrastructure mapping in Norway, the Netherlands and France, where the mapping is quite complete and heights are included.

#### 1. Render different types of structures with appropriate icons
For *tower:type=communication*:
- ![dome](https://raw.githubusercontent.com/gravitystorm/openstreetmap-carto/master/symbols/man_made/telescope_dome.svg) *tower:structure=dome* (or a similar icon with just the dome, not the telescope).
- ![dish](https://raw.githubusercontent.com/gravitystorm/openstreetmap-carto/master/symbols/man_made/tower_dish.svg) *tower:structure=dish*.
- ![freestanding](https://raw.githubusercontent.com/gravitystorm/openstreetmap-carto/master/symbols/man_made/tower_cantilever_communication.svg) *tower:structure=freestanding*.
- ![lattice](https://raw.githubusercontent.com/gravitystorm/openstreetmap-carto/master/symbols/man_made/tower_lattice_communication.svg) *tower:structure=lattice*.
- ![communications_tower](https://raw.githubusercontent.com/gravitystorm/openstreetmap-carto/master/symbols/man_made/communications_tower.svg) *man_made=communications_tower*.
- ![mast](https://raw.githubusercontent.com/gravitystorm/openstreetmap-carto/master/symbols/man_made/mast_communications.svg) *man_made=mast*, *tower:structure=guyed_lattice* and *tower:structure=guyed_tube* (and others).

#### 2. Render telecom towers/masts at different zoom levels according to importance
 - *man_made=communications_tower* visible from zoom 6. They are usually the most important towers in a national link network.
 - *tower:type=communication* + *height>=100* visible from zoom 7.
 - *tower:type=communication* + *height>=60* and *communication:television=yes* visible from zoom 8. There are fewer tv transmitters than mobile masts.
 - *tower:type=communication* + *height>=40* visible from zoom 9.
 - Rest from zoom 10.

**Comment by flacombe** [comment]
Hello

Good points.
A few telecom=* keys has been introduced years ago.

I'm currently rendering them for France only here https://map.infos-reseaux.com/#12.54/45.17688/0.73124/L,T
See style: https://github.com/flacombe/openinframap/blob/latest/web/src/style/style_oim_telecoms.js

It could be great to support telecom=* key as well as radio features.

**Comment by mathiash98** [comment]
I realised that my progress completely stopped because it's not possible to run the project with node v20 or even as low as Node v10 as described in https://github.com/mapbox/spritezero/issues/82.

Maybe we can consider replacing the use of `@mapbox/spritezero` with something else that supports higher version node and is buildable on Mac M1?

**Comment by russss** [comment]
Yeah, I've recently been investigating this for another project. I've created #165 which has my current thinking on this.

**Comment by NKAmapper** [comment]
Imagico recently refined a number of icons which could be useful for this proposal:

man_made=mast:
![](https://imagico.de/blog/wp-content/uploads/2024/10/mast_variants_19_40_x2.png)

man_made=tower:
![](https://imagico.de/blog/wp-content/uploads/2024/10/tower_variants_19_40_x2.png)

Link to blog: https://imagico.de/blog/en/more-map-symbols/
Link to source: https://github.com/imagico/osm-carto-alternative-colors/tree/master/symbols/sources/towers
CC-BY-SA 4.0 license
